recipes/qt/6.x.x/conanfile.py

```diff
@@ -692,34 +723,11 @@ def build(self):
             tools.replace_in_file(f,
                 " IMPORTED)\n",
                 " IMPORTED GLOBAL)\n", strict=False)
-        with tools.vcvars(self) if self._is_msvc else tools.no_op():
-            # next lines force cmake package to be in PATH before the one provided by visual studio (vcvars)
-            build_env = tools.RunEnvironment(self).vars if self._is_msvc else {}
-            build_env["MAKEFLAGS"] = "j%d" % tools.cpu_count()
-            build_env["PKG_CONFIG_PATH"] = [self.build_folder]
-            if self.settings.os == "Windows":
-                if not "PATH" in build_env:
-                    build_env["PATH"] = []
-                build_env["PATH"].append(os.path.join(self.source_folder, "qt6", "gnuwin32", "bin"))
-            if self._is_msvc:
-                # this avoids cmake using gcc from strawberryperl
-                build_env["CC"] = "cl"
-                build_env["CXX"] = "cl"
-            with tools.environment_append(build_env):
-
-                if tools.os_info.is_macos:
-                    tools.save(".qmake.stash" , "")
-                    tools.save(".qmake.super" , "")
+            
+        with self._build_context():
+            cmake = self._configure_cmake()
+            cmake.build()
 
-                cmake = self._configure_cmake()
-                if tools.os_info.is_macos:
-                    tools.save("bash_env", 'export DYLD_LIBRARY_PATH="%s"' % ":".join(RunEnvironment(self).vars["DYLD_LIBRARY_PATH"]))
-                with tools.environment_append({
-                    "BASH_ENV": os.path.abspath("bash_env")
-                }) if tools.os_info.is_macos else tools.no_op():
```

Comment:
Thanks!
I think all the set-up creating files (`.qmake.stash`, `.qmake.super` and `bash_env`) should only happen only in build method, and not in package. Also, there was a reason (which I forgot) for creating `bash_env` after configuring cmake and not before, so I'd prefer if it stayed this way.

Reply:
I was worried about that, too. I should be able to add two different context methods, right?

Reply:
the best it to put only common stuff in the context method, and you do the rest directly in the build method, as it was until now basically.

Reply:
Look better?